File: api/rust/core/src/resp.rs
use crate::env;

pub fn set_header(key: &str, val: &str) {
	env::resp_set_header(key.as_ptr(), key.len() as i32, val.as_ptr(), val.len() as i32);
}

pub fn content_type(ctype: &str) {
	set_header("Content-Type", ctype);
}

File: sdk/tinygo/runnable/resp/resp.go
//go:build tinygo.wasm

package resp

import (
	"github.com/suborbital/sdk/api/tinygo/runnable/internal/ffi"
)

func SetHeader(key, value string) {
	ffi.RespSetHeader(key, value)
}

func ContentType(contentType string) {
	SetHeader("Content-Type", contentType)
}
